src/Common/CurrentThread.cpp

```diff
@@ -110,4 +110,26 @@ ThreadGroupStatusPtr CurrentThread::getGroup()
     return current_thread->getThreadGroup();
 }
 
+MemoryTracker * CurrentThread::getUserMemoryTracker()
+{
+    if (unlikely(!current_thread))
+        return nullptr;
+
+    if (auto group = current_thread->getThreadGroup())
```

Comment:
Maybe we should for `current_thread->memory_tracker` call `getParent()` till we get a tracker with level = `User`?
Just to support some strange scenario when we have shorter chain, like `Thread -> User -> Global` (which can probably be for some bg operations).